### PR TITLE
Fix the scroll to manage fragment

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-5899

### Change description ###
Adding in ability via code to direct the scroll to the fragment
Note: Should be being done by browser - this is an applied fix that can work in tandem with any future required fix

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by tarikdogan-hmcts — APPROVED** [review]

File: src/work-allocation-2/containers/work-case-list/work-case-list.component.ts [hunk 1]
```diff
@@ -73,6 +73,17 @@ export class WorkCaseListComponent implements OnChanges {
...
+      if (s instanceof NavigationEnd) {
+        const tree = this.router.parseUrl(this.router.url);
+        if (tree.fragment) {
+          const element = document.querySelector('#' + tree.fragment);
+          if (element) {
+            element.scrollIntoView({block: "center"});
+          }
+        }
+      }
+    })
```
Comment (udaydenduluri33) [inline src/work-allocation-2/containers/work-case-list/work-case-list.component.ts:86]: semicolon is missing

Comment (connorpgpmcelroy) [reply src/work-allocation-2/containers/work-case-list/work-case-list.component.ts:86]: Ok, I'll add

File: src/work-allocation-2/containers/task-list/task-list.component.ts [hunk 2]
```diff
@@ -76,6 +76,17 @@ export class TaskListComponent implements OnChanges {
...
+      if (s instanceof NavigationEnd) {
+        const tree = this.router.parseUrl(this.router.url);
+        if (tree.fragment) {
+          const element = document.querySelector('#' + tree.fragment);
+          if (element) {
+            element.scrollIntoView({block: "center"});
+          }
+        }
+      }
+    })
```
Comment (udaydenduluri33) [inline src/work-allocation-2/containers/task-list/task-list.component.ts:89]: semicolon missing

File: src/work-allocation-2/containers/task-list/task-list.component.ts [hunk 3]
```diff
@@ -76,6 +76,17 @@ export class TaskListComponent implements OnChanges {
   }
 
   public ngOnChanges(): void {
+    this.router.events.subscribe(s => {
+      if (s instanceof NavigationEnd) {
```
Comment (udaydenduluri33) [inline src/work-allocation-2/containers/task-list/task-list.component.ts:80]: can we give `s` a better name?

File: src/work-allocation-2/containers/work-case-list/work-case-list.component.ts [hunk 4]
```diff
@@ -73,6 +73,17 @@ export class WorkCaseListComponent implements OnChanges {
   }
 
   public ngOnChanges(): void {
+    this.router.events.subscribe(s => {
+      if (s instanceof NavigationEnd) {
+        const tree = this.router.parseUrl(this.router.url);
```
Comment (udaydenduluri33) [inline src/work-allocation-2/containers/work-case-list/work-case-list.component.ts:78]: can we give `s` a better name?

**Review by tarikdogan-hmcts — APPROVED** [review]

**Review by udaydenduluri33 — APPROVED** [review]
Approved, Thanks for fix

**Pull request closed without merging**